apps/files/src/components/Collaborators/EditCollaborator.vue

```diff
@@ -55,13 +60,34 @@ export default {
     return {
       selectedRole: null,
       additionalPermissions: null,
-      saving: false
+      saving: false,
+      expirationDate: null,
+      // Since expiration date can ba reset to null we can't depend on it when checking for changes
+      expirationDateResetted: false
```

Comment:
another pattern would be storing the old value and new value.
it does seem like the original value is available in `originalExpirationDate()`, why not use it ?